src/hotspot/share/utilities/globalDefinitions.hpp

```diff
@@ -25,6 +25,7 @@
 #ifndef SHARE_UTILITIES_GLOBALDEFINITIONS_HPP
 #define SHARE_UTILITIES_GLOBALDEFINITIONS_HPP
 
+#include "metaprogramming/primitiveConversions.hpp"
```

Comment:
I think this should not be done, for the same reasons as not including the suggested bit_cast
(https://git.openjdk.org/jdk/pull/11865).  There is substantial overlap between this proposed
change and that one, though I think the bit_cast proposal is more problematic.

Rather than adding a new dependency here, separate out the code that uses the conversions
into new files and update the (much smaller number than use globalDefinitions.hpp) users
of those functions to include the new headers.

Reply:
I previously updated the bit_cast PR to just be `PrimitiveConversions::cast` implementation, but using __builtin_bit_cast when available. So this and that are basically the same.

Reply:
Dependencies to `globalDefinitions` and `primitiveCopnversions` are broken down to smaller header files and all the dependencies in other headers are addressed correspondingly.